src/main/java/edu/depaul/cdm/se452/d2l_mock/discussion_thread/DiscussionThreadController.java

```diff
@@ -2,24 +2,35 @@
 
 import java.util.List;
 
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.stereotype.Controller;
 import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.ui.Model;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.PostMapping;
 import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.cache.annotation.Cacheable;
 
-@RestController
-@RequestMapping("/api/discussion-threads")
+@Controller
+@RequestMapping("/discussion-threads")
 public class DiscussionThreadController {
     @Autowired
     private DiscussionThreadService service;
 
     @GetMapping
     @Cacheable(value = "discussion-threads")
-    public List<DiscussionThread> list() {
-        return service.list();
+    public String list(Model model) {
+        List<DiscussionThread> threads = service.list();
+        model.addAttribute("threads", threads);
```

Comment:
"threads" seems to be a mismatch with thread in discussionThreads.html. Changing to "thread" here passed the test.

Reply:
![image](https://github.com/user-attachments/assets/b1c5f642-01f2-47e1-a203-7ed8ed73b2fc)

Reply:
![image](https://github.com/user-attachments/assets/9cd8fc90-7884-4db7-b990-af3460e4ab20)

Reply:
Looks like the binding for the entire arraylist works with that change.

Reply:
I'm not sure if the entire list is being handled. I'm not familiar with BindingAwareModelMap